faster_whisper/transcribe.py

```diff
@@ -731,6 +735,8 @@ def generate_with_fallback(
             decode_result = max(
                 below_cr_threshold_results or all_results, key=lambda x: x[1]
             )
+            # to pass final temperature for prompt_reset_on_temperature
+            WhisperModel.fallbacks_final_temperature = temperature
```

Comment:
Handling this with a global variable is not very clean.
Is there an issue with simply replacing decode_result temperature with the accurate temperature ?

Reply:
> Handling this with a global variable is not very clean.

It's not a global, it's local, a class variable. 
To my unprofessional eye this looked cleanest as other ways looked too intrusive.

> Is there an issue with simply replacing decode_result temperature with the accurate temperature ?

I don't know how to replace it, if you know then please make a proper PR of this fix.

Reply:
decode_result = (decode_result[0], decode_result[1], 1, decode_result[3])

Reply:
Thx. Redone it.